test/integration/integration_test.exs

```diff
@@ -20,16 +36,99 @@ defmodule Membrane.MP3.Lame.Encoder.IntegrationTest do
     assert a == b
   end
 
-  test "Encode raw samples" do
-    in_path = "../fixtures/input.pcm" |> Path.expand(__DIR__)
-    out_path = "/tmp/output-lame.mp3"
-    ref_path = "../fixtures/ref.mp3" |> Path.expand(__DIR__)
-    on_exit(fn -> File.rm(out_path) end)
+  defp assert_correct_sink_buffers_pts(
+         raw_pipeline,
+         encoded_pipeline,
+         raw_to_encoded_buffers_ratio
+       ) do
+    receive do
+      {Membrane.Testing.Pipeline, ^raw_pipeline,
+       {:handle_child_notification, {{:buffer, %Buffer{pts: raw_pts}}, :sink_raw}}} ->
+        encoded_pts =
+          case raw_to_encoded_buffers_ratio do
+            :one_to_one ->
+              assert_sink_buffer(encoded_pipeline, :sink_encoded, %Buffer{pts: encoded_pts})
+              encoded_pts
+
+            :one_to_two ->
+              assert_sink_buffer(encoded_pipeline, :sink_encoded, %Buffer{pts: encoded_pts})
+              assert_sink_buffer(encoded_pipeline, :sink_encoded, _skip_buffer)
+              encoded_pts
+
+            :two_to_one ->
+              assert_sink_buffer(encoded_pipeline, :sink_encoded, %Buffer{pts: encoded_pts})
+              assert_sink_buffer(raw_pipeline, :sink_raw, _skip_buffer)
+              encoded_pts
+          end
+
+        assert raw_pts == encoded_pts
+
+        assert_correct_sink_buffers_pts(
+          raw_pipeline,
+          encoded_pipeline,
+          raw_to_encoded_buffers_ratio
+        )
+    after
+      0 ->
+        :ok
```

Comment:
This may finish before the stream ends when a subsequent buffer doesn't arrive at the mailbox on time. Better exit when the end of stream arrives.

Reply:
before this receive block we always do `assert_end_of_stream`, so I don't think that could be the case